finetrainers/trainer.py

```diff
@@ -375,6 +391,11 @@ def prepare_trainable_parameters(self) -> None:
 
         # TODO(aryan): handle torch dtype from accelerator vs model dtype; refactor
         self.state.weight_dtype = weight_dtype
+        if self.args.mixed_precision != _INVERSE_DTYPE_MAP[weight_dtype]:
+            logger.info(
+                f"`mixed_precision` was set to {_INVERSE_DTYPE_MAP[weight_dtype]} which different from what was initially passed ({self.args.mixed_precision})."
```

Comment:
```suggestion
            logger.warning(
                f"`mixed_precision` was set to {_INVERSE_DTYPE_MAP[weight_dtype]} which is different from configured argument ({self.args.mixed_precision})."
```

Reply:
Actually, on second thought, I don't think we should support `mixed_precision` as an argument or handle it from accelerate at all. My reasoning is:
- We provide the option to set `transformer_dtype` via the CLI args. It will load the transformer in requested dtype. If not provided, we default to the sensible value
- We are first the transformer in some dtype (either by CLI arg or using sensible default), which might be different from what accelerate was configured with. That will cause lossy conversion and just make the transformer a lot more useless to start training with
- Many of the latest transformers only work in one dtype - either bfloat16 or float16. We don't handle gradient scaling related stuff properly yet, so float16 training of transformer is most likely not correct IMO. Only bfloat16 will work sensibly at the moment

We should configure the dtypes as recommended defaults from authors, and raise a warning if it is changed by user so it is clear that:
- they know what they are doing, or
- they are making a mistake

Only if a transformer was trained in pure fp32 will it be able to support fp16 and bf16, but that may not always be true. My concern is to address the fact that most transformers these days are simply trained in bf16 and very much incompatible with fp16, so it makes no sense to support this option

Reply:
Okay with the general thoughts but some observations:

* SDXL was trained with BF16 (unet) but fine-tunes fine with FP16 (both LoRA and non-LoRA). So, it depends on model to model. Same goes for Flux, SD3, etc. 
* We don't have to explicitly handle gradient scaling if the `mixed_precision` passed to `Accelerator` is working sensibly with the model under consideration. IIUC `accelerate` should handle this for us already as behind the scenes it runs AMP too. 
* Another problem is `bfloat16` won't work on 3090 24GB machines. So, in this case, users should configure the `transformer_dtype` argument. 
* Users may want to keep the `transformer` in FP32 but train using a certain mixed-precision scheme but this can be revised later.

Reply:
> SDXL was trained with BF16 (unet) but fine-tunes fine with FP16 (both LoRA and non-LoRA). So, it depends on model to model. Same goes for Flux, SD3, etc.

Yes, so this is why let's do this:
- Always use sensible defaults based on recommendations so that if user does not provide any custom dtype themselves, it will default to what works. I believe this is already the case
- The video models are more sensitive to dtype and usually only work with one dtype (usually bf16 because they are almost always trained in that precision). For example, CogVideoX-5b models will result in garbage if trained on fp16. What I was asking above was to:
  - Allow the option for user to specify custom dtype for different components (we do this already)
  - Remove `mixed_precision` related arguments completely because it is now confusing. What happens if someone chooses bf16 as transformer dtype but sets mixed_precision to fp16? We will load transformer as bf16 and then cast to fp16, which is lossy
  - For this to not happen, what we could do is raise a warning if defaulted sensible dtype does not match dtype specified by user (some logic for this per model when validating args)

We can proceed here with how it is, but this will be the cause for confusion almost certainly I think so I will take up the removing `mixed_precision` part in a follow-up PR

> Users may want to keep the transformer in FP32 but train using a certain mixed-precision scheme but this can be revised later.

I highly doubt this will be the case. Most, if not all, video models I've worked on integrating don't work on more than one dtype (even an image model like Sana), and are trained in bf16, so upcasting to fp32 already makes it a bit unsuitable to replicating training environment. Training in FP32 is also atleast 4-8x slower right off the bat, while using an absurd amount of memory (> 2x of bf16) last I tested when I started adding Cog scripts. If required, let's revisit it later.

Reply:
Yes I think we are saying the same things here. No problem on my end.